recipes-security/refpolicy/refpolicy-mcs-2.%/policy/modules/services/rpcbroker.te

```diff
@@ -0,0 +1,99 @@
+#############################################################################
+#
+# Copyright (C) 2018 Assured Information Security, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+#############################################################################
+
+policy_module(rpcbroker, 0.1)
+
+# Type for rpc-broker executable file.
+type rpcbroker_exec_t;
+
+# Type for rpc-broker.rules file.
+type rpcbroker_etc_t;
+files_type(rpcbroker_etc_t)
+
+type rpcbroker_t;
+type rpcbroker_websockets_t;
+
+init_daemon_domain(rpcbroker_t, rpcbroker_exec_t)
+
+allow rpcbroker_t rpcbroker_etc_t:file rw_file_perms;
+allow rpcbroker_t self:fifo_file rw_fifo_file_perms;
+allow rpcbroker_websockets_t rpcbroker_etc_t:file rw_file_perms;
+allow rpcbroker_websockets_t self:fifo_file rw_fifo_file_perms;
+
+##
+allow rpcbroker_t rpcbroker_exec_t:file execute;
+
+allow rpcbroker_t self:unix_dgram_socket { sendto create_socket_perms };
+allow rpcbroker_websockets_t self:unix_dgram_socket { sendto create_socket_perms };
+
+init_daemon_domain_spec_domtrans(rpcbroker_websockets_t, rpcbroker_exec_t)
+
+xc_files_rw_v4v_chr(rpcbroker_t)
+xc_files_rw_v4v_chr(rpcbroker_websockets_t)
+
+# Allow xenstore in dom0
+xen_stream_connect_xenstore(rpcbroker_t)
+xen_stream_connect_xenstore(rpcbroker_websockets_t)
+# Allow xenstore outside of dom0
+dev_rw_xen(rpcbroker_t)
+dev_rw_xen(rpcbroker_websockets_t)
+
+dev_write_urand(rpcbroker_t)
```

Comment:
I believe this is something we had to allow because of how libwebsockets initializes its connections.

https://libwebsockets.org/git/libwebsockets/tree/lib/roles/ws/client-ws.c?h=v3.1-stable#n117